packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -490,95 +493,154 @@ export function commitPassiveEffectDurations(
   }
 }
 
-function commitLifeCycles(
+function commitLayoutEffectOnFiber(
   finishedRoot: FiberRoot,
   current: Fiber | null,
   finishedWork: Fiber,
   committedLanes: Lanes,
 ): void {
-  switch (finishedWork.tag) {
-    case FunctionComponent:
-    case ForwardRef:
-    case SimpleMemoComponent: {
-      // At this point layout effects have already been destroyed (during mutation phase).
-      // This is done to prevent sibling component effects from interfering with each other,
-      // e.g. a destroy function in one component should never override a ref set
-      // by a create function in another component during the same commit.
-      if (
-        enableProfilerTimer &&
-        enableProfilerCommitHooks &&
-        finishedWork.mode & ProfileMode
-      ) {
-        try {
-          startLayoutEffectTimer();
+  if ((finishedWork.flags & (Update | Callback)) !== NoFlags) {
```

Comment:
Regarding the diff in this file, Git got confused by an indentation change. I left some comments with explanations, but accidentally put  them on the next PR instead of this one. Basically, it's just whitespace: https://github.com/facebook/react/pull/20596#discussion_r557928847